apps/src/calc/inputIterator.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Given a set of values (i.e. [1,2,3], and a number of parameters, generates
+ * all possible combinations of values.
+ */
+var InputIterator = function (values, numParams) {
```

Comment:
it could be, but I'm (a) a little skeptical we'll end up needing this anywhere else and (b) wary of reducing specificity without yet having another use case.